lightning/src/chain/channelmonitor.rs

```diff
@@ -1331,19 +1331,19 @@ impl<Signer: EcdsaChannelSigner> ChannelMonitor<Signer> {
 		ChannelMonitor { inner: Mutex::new(imp) }
 	}
 
-	pub(crate) fn new(secp_ctx: Secp256k1<secp256k1::All>, keys: Signer, shutdown_script: Option<ScriptBuf>,
+	pub(crate) fn new(secp_ctx: Secp256k1<secp256k1::All>, mut keys: Signer, shutdown_script: Option<ScriptBuf>,
 	                  on_counterparty_tx_csv: u16, destination_script: &Script, funding_info: (OutPoint, ScriptBuf),
 	                  channel_parameters: &ChannelTransactionParameters, holder_pays_commitment_tx_fee: bool,
 	                  funding_redeemscript: ScriptBuf, channel_value_satoshis: u64,
 	                  commitment_transaction_number_obscure_factor: u64,
 	                  initial_holder_commitment_tx: HolderCommitmentTransaction,
 	                  best_block: BestBlock, counterparty_node_id: PublicKey, channel_id: ChannelId,
 	) -> ChannelMonitor<Signer> {
+	
+		keys.provide_channel_parameters(channel_parameters);
```

Comment:
I move the `provide_channel_parameters` call to here because there were some tests that built a `ChannelMonitor` without first calling `provide_channel_parameters` on the `keys: Signer`.

Further below in `initial_commitment_signed`, I delete the `provide_channel_parameters` call, as it is now called here.

Reply:
We now need to call `provide_channel_parameters` before `ChannelMonitor::new` because `ChannelSigner::get_counterparty_payment_script` assumes that `provide_channel_parameters` has already been called - see the doc for `ChannelSigner::get_counterparty_payment_script`.